include/revng/Support/GraphAlgorithms.h

```diff
@@ -518,26 +541,844 @@ nodesBetweenNewImpl(GraphT Source,
 }
 
 template<class GraphT>
-inline llvm::SmallPtrSet<GraphT, 4>
+inline SmallSetVector<GraphT>
 nodesBetweenNew(GraphT Source,
                 GraphT Destination,
-                const llvm::SmallPtrSetImpl<GraphT> *IgnoreL = nullptr) {
+                const SmallSetVector<GraphT> *IgnoreList = nullptr) {
   return nodesBetweenNewImpl<GraphT, llvm::GraphTraits<GraphT>>(Source,
                                                                 Destination,
-                                                                IgnoreL);
+                                                                IgnoreList);
 }
 
 template<class GraphT>
-inline llvm::SmallPtrSet<GraphT, 4>
+inline SmallSetVector<GraphT>
 nodesBetweenNewReverse(GraphT Source,
                        GraphT Destination,
-                       const llvm::SmallPtrSetImpl<GraphT> *IgnoreL = nullptr) {
+                       const SmallSetVector<GraphT> *IgnoreList = nullptr) {
   using namespace llvm;
   return nodesBetweenNewImpl<GraphT, GraphTraits<Inverse<GraphT>>>(Source,
                                                                    Destination,
-                                                                   IgnoreL);
+                                                                   IgnoreList);
+}
+
+template<class NodeT>
+bool intersect(SmallSetVector<NodeT> &First, SmallSetVector<NodeT> &Second) {
+  // TODO: Upon migration to LLVM 17, we should use the `llvm::set_intersection`
+  // provided in `SetOperations.h` header file.
+  llvm::SmallPtrSet<NodeT, 4> Intersection(First.begin(), First.end());
+  llvm::set_intersect(Intersection, Second);
+  return !Intersection.empty();
+}
+
+template<class NodeT>
+bool subset(SmallSetVector<NodeT> &Contained,
+            SmallSetVector<NodeT> &Containing) {
+  return llvm::set_is_subset(Contained, Containing);
+}
+
+template<class NodeT>
+bool equal(SmallSetVector<NodeT> &First, SmallSetVector<NodeT> &Second) {
+  return llvm::set_is_subset(First, Second)
+         and llvm::set_is_subset(Second, First);
+}
+
+template<class NodeT>
+bool simplifyRegionsStep(llvm::SmallVector<SmallSetVector<NodeT>> &R) {
+  for (auto RegionIt1 = R.begin(); RegionIt1 != R.end(); RegionIt1++) {
+    for (auto RegionIt2 = std::next(RegionIt1); RegionIt2 != R.end();
+         RegionIt2++) {
+      bool Intersects = intersect(*RegionIt1, *RegionIt2);
+      bool IsIncluded = subset(*RegionIt1, *RegionIt2);
+      bool IsIncludedReverse = subset(*RegionIt2, *RegionIt1);
+      bool AreEquivalent = equal(*RegionIt1, *RegionIt2);
+      if (Intersects
+          and (((!IsIncluded) and (!IsIncludedReverse)) or AreEquivalent)) {
+        (*RegionIt1).insert((*RegionIt2).begin(), (*RegionIt2).end());
```

Comment:
`(*x).insert` -> `x->insert`

Reply:
Right, changed in 7b60c49bba5376fbe0c3de3d0f15e2c1c810e9c7.